tests/Repository/GeneratedBookRepositoryTest.php

```diff
@@ -22,17 +22,18 @@ class GeneratedBookRepositoryTest extends KernelTestCase {
 	protected function setUp(): void {
 		parent::setUp();
 		self::bootKernel();
-		$this->entityManager = self::$container->get( 'doctrine' )->getManager();
+		$this->entityManager = self::getContainer()->get( 'doctrine' )->getManager();
 		$this->genBookRepo = $this->entityManager
 			->getRepository( GeneratedBook::class );
 	}
 
 	/**
-	 * @covers GeneratedBookRepository::getTypeAndLangStats
+	 * @covers \App\Repository\GeneratedBookRepository::getTypeAndLangStats
 	 */
 	public function testAddAndRetrieve() {
 		// Make sure it's empty to start with.
-		static::assertEmpty(
+		static::assertSame(
```

Comment:
for me: Is there a difference between `static::assertSame` and `$this->assertSame`? I would guess they are the same function and we should harmonize on something?

Reply:
Good point. [The docs](https://docs.phpunit.de/en/11.5/assertions.html#appendixes-assertions) say that there's no right way, and as we've got more `$this->` than `static` I'll stick to that! And also use `assertSame` over `assertEquals` unless we want loose checking (which it doesn't look like we do anywhere).